app/src/main/java/com/github/h3lp3rs/h3lp/messaging/Conversation.kt

```diff
@@ -179,12 +150,98 @@ class Conversation(
         database.delete(conversationId)
     }
 
+
+    /**
+     * Load the cache if exist
+     * @warning should be called in onCreate method of the activity
+     */
+    fun loadCache() {
+        val storage = storageOf(Storages.MSG_CACHE)
+        val cache: HashMap<String, String> = storage.getObjectOrDefault(
+            conversationId,
+            HashMap::class.java,
+            HashMap<String, String>()
+        ) as HashMap<String, String>
+        allMessages.putAll(cache)
+    }
+
+
+    /**
+     * Save the cache for the conversation
+     * @warning should be called in the onPause method of the activity
+     */
+    fun saveCache() {
+        Storages.MSG_CACHE.setOnlineSync(false)
+        val storage = storageOf(Storages.MSG_CACHE)
```

Comment:
You could create a variable messageCache as it is used twice in this scope and also import statically MSG_CACHE